### ISSUE TITLE
Conditional Generators

### ISSUE THREAD

**Issue by doflo-dfa** [prisma/prisma#5413]
## Problem

It would be useful to be able to conditionally run generators based on config or environment. In our scenario the issue is that our schema is large and the userland class generator is very slow like 5 - 6 minutes slow. 

In reality we do not need to run this generator after every change so we have cheated by using the env() helper method to pull in and out the path of the generators we need and when, a good example of this is in our CI process where we want prisma client to be generated but we do not want ( and can not due to the compile and environment issues it would cause ) our classes that we check into source control to be regenerated

The solution we have today works but it is not ideal, first of all the schema validation does not like what we have done, it works and compiles but it is not happy about it.

## Suggested solution

I would think a config file that supported profiles would solve the problem, ideally separating out the schema from the generators so you don't have to keep them in sync.

## Alternatives

An alternative approach would be an "enabled" attribute for the generators and then modifying the schema validation to allow the env() method to be used in more places than just the database string.

**Comment by joshg47** [comment]
Yep, this would make custom generators a lot more useful. Bls add soon

**Comment by gabrieljoelc** [comment]
A concrete example for me is I want to update a SVG diagram (using [prisma-erd-generator](https://github.com/keonik/prisma-erd-generator)) from the schema for every PR that the schema updates but I only want the types to update and not the diagram in the CI for the `main` branch.

**Comment by janpio** [comment]
For now we have a CLI parameter `--generator` which you can use to run them selectively: https://www.prisma.io/docs/reference/api-reference/command-reference#arguments-1 With that at leas you should be able to script your own way to a solution.
